src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParameterResourceDataStore.cs

```diff
@@ -0,0 +1,93 @@
+        private readonly Func<IScoped<ISearchService>> _searchServiceFactory;
+        private readonly IModelInfoProvider _modelInfoProvider;
+        private readonly ILogger _logger;
+
+        public SearchParameterResourceDataStore(
+            ISearchParameterOperations searchParameterOperations,
+            Func<IScoped<ISearchService>> searchServiceFactory,
+            IModelInfoProvider modelInfoProvider,
+            ILogger<SearchParameterResourceDataStore> logger)
+        {
+            EnsureArg.IsNotNull(searchParameterOperations, nameof(searchParameterOperations));
+            EnsureArg.IsNotNull(searchServiceFactory, nameof(searchServiceFactory));
+            EnsureArg.IsNotNull(modelInfoProvider, nameof(modelInfoProvider));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _searchParameterOperations = searchParameterOperations;
+            _searchServiceFactory = searchServiceFactory;
+            _modelInfoProvider = modelInfoProvider;
+            _logger = logger;
+        }
+
+        public async Task EnsureInitialized()
+        {
+            // now read in any previously POST'd SearchParameter resources
+            using (var search = _searchServiceFactory())
+            {
+                string continuationToken = null;
+                do
+                {
+                    var queryParams = new List<Tuple<string, string>>();
+                    if (continuationToken != null)
+                    {
+                        queryParams.Add(new Tuple<string, string>(KnownQueryParameterNames.ContinuationToken, continuationToken));
+                    }
+
+                    var result = await search.Value.SearchAsync("SearchParameter", queryParams, cancellationToken: default);
+                    if (!string.IsNullOrEmpty(result?.ContinuationToken))
+                    {
+                        continuationToken = Convert.ToBase64String(System.Text.Encoding.UTF8.GetBytes(result.ContinuationToken));
+                    }
+                    else
+                    {
+                        continuationToken = null;
+                    }
+
+                    if (result != null && result.Results != null && result.Results.Count() > 0)
+                    {
+                        var searchParams = result.Results.Select(r => r.Resource.RawResource.ToITypedElement(_modelInfoProvider)).ToList();
+
+                        foreach (var searchParam in searchParams)
+                        {
+                            try
+                            {
+                                await _searchParameterOperations.AddSearchParameterAsync(searchParam);
+                            }
+                            catch (SearchParameterNotSupportedException ex)
+                            {
+                                _logger.LogError(ex, $"Error loading search parameter {searchParam.GetStringScalar("url")} from data store.");
```

Comment:
>LogError [](start = 40, length = 8)

Would this resolve into exception in geneva logs?
If yes, can we lower priority of this message?
Since there is no way to tell about this errors to customer we gonna get them on every service load, which can be overwhelming, and there is no action needed from our side.

I would rather have current messages as warning or informational, and had final `catch (Exception ex)` which would log error.
Generic unexpected error is something we need to worry about.

Reply:
Agreed, lowered to warning.